model/src/pyrenew/metaclass.py

```diff
@@ -94,6 +94,46 @@ def _assert_sample_and_rtype(
     return None
 
 
+class SampledValue:
+    """
+    A container for a sampled value from a RandomVariable.
+    """
+
+    def __init__(
+        self,
+        value: ArrayLike | None,
+        t_start: int | None = None,
+        t_unit: int | None = None,
+    ) -> None:
+        """
+        Default constructor for SampledValue
+
+        Parameters
+        ----------
+        value : ArrayLike
+            The sampled value.
+        t_start : int, optional
+            The start time of the value.
+        t_unit : int, optional
+            The unit of time relative to the model's fundamental (smallest) time unit.
+
+        Returns
+        -------
+        None
+        """
+
+        if value is not None:
+            assert isinstance(
+                value, ArrayLike
+            ), "value should be an array-like object."
```

Comment:
Are we sure we want this (versus arbitrary `pytrees`?

Reply:
Good question. I would say yes, keep things as ArrayLike objects, but pytrees should be, in principle, OK. Since I'm making the SampledValue a namedtuple, I'm taking this out and we can leave it for a later discussion if we want to enforce a type.